### ISSUE TITLE
autoformatting

### ISSUE THREAD

**Issue by madagaluna** [susanBuck/e2-fall23#14]
I think I have autoformatting on but I am concerned because my previous homework was not formatted correctly throughout.    Preferences/settings takes me to **settings.json** where "editor.formatOnSave": true," is set so I am not sure why my formatting wasn't corrected when I saved it.  Any thoughts?  Should I be landing on **settings.json** when I go to my settings?

Thanks in advance,

Mary

**Comment by susanBuck** [comment]
Hi Mary,

Sounds like your auto-formatting settings are correct. Might be a missing extension on the server?

While connected to your server in VSCode, can you run the command `code --list-extensions` and share the output so I can double check your extension list?

For example, here's what the output looks like on my server:

<img width="670" alt="image" src="https://github.com/susanBuck/e2-fall23/assets/237711/c02c1915-2c05-4b43-a03f-3f3bb0374835">

**Comment by madagaluna** [comment]
Thanks for the example!   I get the same output.    Any other thoughts on what it may be?

As for a very aspirational question, would it be acceptable to use this code <https://www.codexworld.com/distance-between-two-addresses-google-maps-api-php/> in my final game?  My goal would be to compare the times and locations performers want to perform in building the schedule.  Again, very aspirational - just getting the code to work for times, let alone creating functions and forms (and which forms!) is keeping me more than entertained.

Thanks,
Mary
[https://www.codexworld.com/wp-content/uploads/2015/06/calculate-distance-between-two-addresses-google-maps-api-php-codexworld.png]<https://www.codexworld.com/distance-between-two-addresses-google-maps-api-php/>
Distance Between Two Addresses using Google Maps API and PHP - CodexWorld<https://www.codexworld.com/distance-between-two-addresses-google-maps-api-php/>
Calculate the distance between two addresses using Google Maps Geocoding API and PHP - Get distance between two addresses through PHP. Example code to calculate distance using PHP and Google Map API.
www.codexworld.com



________________________________

**Comment by madagaluna** [comment]
![output](https://github.com/susanBuck/e2-fall23/assets/78892871/ce3430c4-1478-4c74-98cb-2a6e8229a58f)

**Comment by susanBuck** [comment]
Ok, extensions all look good.

I logged in to your server to test things out first hand and I found the auto-formatter was working.

For example, I opened your file `/e2/p1/index.php` and here's what it looked like initially:

<img width="1047" alt="image" src="https://github.com/susanBuck/e2-fall23/assets/237711/811b92e4-f103-4817-8cf8-c6c8579d93a2">

Then I saved the file and it changed to look like this:

<img width="1048" alt="image" src="https://github.com/susanBuck/e2-fall23/assets/237711/67996111-9c33-46c4-822b-701d26284808">

Differences include:
+ Everything in the for loop, including comments, is now correctly indented 1x
+ The start and close curly brackets of the if/else conditional are moved to the correct location

I reverted the changes when I was done testing. Can you open `/e2/p1/index.php` and do a save and see if you see the same results? 

---

As for using the outside API - yes it would be acceptable for incorporate code like that into a project, but I would approach it in one of the following ways:

Option 1)
Don't make the project dependent on it. Build something that meets the project requirements without the API first, and then time permitting, add the API-specific feature. This way you make sure you hit the project goals without getting lost in working with the API.

Option 2)
Set a time limit for yourself to play around with the API code (30 minutes or an hour) and if after that time it's not working as expected, table it. This will also help you from falling into a wormhole of working with the API to the detriment of meeting the base project requirements. :)

**Comment by anp2937** [comment]
I also have some issues with making this extension work properly. I installed both client/server versions on extension and when saving the file on Mac with `Command+S` I receive this error:

`PHP CS Fixer: executablePath not found. Try setting "php-cs-fixer.executablePath": "${extensionPath}/php-cs-fixer.phar" and try again.`

Basically I just installed these extensions from the Visual Studio code and did not perform any additional configuration.
Looks like this is something very trivial but I was not able to find the solution. Hope someone from the class had the same issue and found the solution.

**Comment by madagaluna** [comment]
My issue continues and does not resolve with explicitly saving the file.   I am using a PC with Windows 11.

Sent from my iPhone

On Oct 24, 2023, at 1:47 AM, AndriiP ***@***.***> wrote:

﻿

I also have some issues with making this extension work properly. I installed both client/server versions on extension and when saving the file on Mac with Command+S I receive this error:

PHP CS Fixer: executablePath not found. Try setting "php-cs-fixer.executablePath": "${extensionPath}/php-cs-fixer.phar" and try again.

Basically I just installed these extensions from the Visual Studio code and did not perform any additional configuration.
Looks like this is something very trivial but I was not able to find the solution. Hope someone from the class had the same issue and found the solution.

**Comment by susanBuck** [comment]
Thanks for checking again and reporting back @madagaluna. 

Can you share the full contents of your settings.json file? I will put your settings in place on my system to try and recreate the problem and see if we can get to the bottom of this. 

@anp2937 - Just in case other students are having the same problem you're describing it I want to make sure they see this issue, so I created its own thread here: https://github.com/susanBuck/e2-fall23/issues/17

**Comment by susanBuck** [comment]
Quick update: I connected to my server on my Windows machine and was able to replicate the problem: Format on save is activated, yet it's not formatting on save. So - making progress! Will reply again shortly with more details.

**Comment by madagaluna** [comment]
![settings](https://github.com/susanBuck/e2-fall23/assets/78892871/82d3b55c-3194-40da-9050-02cee920d040)
I just added the bottom two settings but they don't fix the issue.  I'll see what you discover.
Thanks!

**Comment by susanBuck** [comment]
I found these settings fixed the issue:

```
 "php-cs-fixer.executablePath": "${extensionPath}/php-cs-fixer.phar",
    "[php]": {
        "editor.formatOnSave": true,
        "editor.defaultFormatter": "junstyle.php-cs-fixer"
    }
```

Looking back in my archive of notes, I used to have instructions that had students add these settings. I'm trying to figure out why I ever removed them since they're clearly necessary.

Let me know if that does the trick for you, @madagaluna, and thanks for the help/input with debugging this.

**Comment by madagaluna** [comment]
That did it.  Thank you!
